File: GVO Code v3/Sync.pas
unit Sync;
interface
uses utils, sysutils,windows,dialogs;
procedure SyncScope;
implementation
uses Main,calccoord,GetQandYU, OMS68SERMC ;
//
//   updated for gvo fork mount 6/1/08
//
//    if shared mem 'sync' coordinates are not zero, then load motor controller with
//        proper counts for that ra, dec
//    enter, shared mem contains current coord. dec, ra
//    get q and y gets quadrant,ha,alt,xcount,ycount and ypole of current ra,dec
//
//

procedure SyncScope;
Var
  CmdStr,CmdStr2,PreStrng : String;
  Quad : integer;
  Alt,HA,Xcount,Ycount : real;
Begin
  If((MovingRA) OR (MovingDec)) Then Exit;
  GetQandY(Coord(pndomem^).RASync,Coord(pndomem^).decSync,Alt,HA,xcount,ycount,quad,Ypole);
  if(Alt < 0 ) then
  begin
     showmessage('Sync below Horizon');
     halt;
  end;
  if(Xcount < 0 ) Then PreStrng := 'AX LP-'       {Send RA Position}
                  Else PreStrng := 'AX LP' ;
  Xcount := abs(xcount);
  Str(Xcount:14:0,CmdStr);
  CmdStr := TrimLeft(CmdStr);
  CmdStr := PreStrng + CmdStr +';';
  SendString(CommRecord,pchar(CmdStr));
  Str(TrkRate:10:6,CmdStr2);
  CmdStr :='AX JF'+CmdStr2 +';';     //Back to Tracking
  SendString(CommRecord,pchar(CmdStr));    {RA Back To TRACKING}
//
  if(Ycount < 0 ) Then PreStrng := 'AY LP-'
                   Else PreStrng := 'AY LP' ;
  YCount := abs(Ycount);
  Str(YCount:14:0,CmdStr);
  CmdStr := TrimLeft(CmdStr);
  CmdStr := PreStrng + CmdStr+';' ;
  SendString(CommRecord,pchar(CmdStr));
//
  Coord(pndomem^).RASync := 0.0 ;                     //reset request
  Coord(pndomem^).decSync := 0.0 ;
  UpdateCoord;
End;
end.

File: GVO Code C++ v1/Sync.cpp
#include "Sync.h"
#include "CommUtils.h"
#include "CalcCoord.h"
#include "OMS68SERMC.h"
#include "GlobalValues.h"
#include "GetQandYU.h"
#include "utils.h"
#include <windows.h>
#include <cmath>
#include <iostream>
#include <sstream>
#include <iomanip>

void SyncScope() {
	if (movingRA || movingDEC)
		return;

	Coord* coord = CommUtils::GetCoordPtr();
	if (!coord) {
		std::cerr << "SyncScope: Coord is null\n";
		return;
	}

	double Alt, HA, Xcount, Ycount;
	int quad;

	GetQandY(coord->RASync, coord->DecSync, Alt, HA, Xcount, Ycount, quad, yPole);

	if (Alt < 0.0) {
		std::cerr << "Sync below Horizon\n";
		return;
	}

	std::string CmdStr, PreStrng;

	// RA position load
	if (Xcount < 0.0)
		PreStrng = "AX LP-";
	else
		PreStrng = "AX LP";

	Xcount = std::abs(Xcount);

	std::ostringstream raStream;
	raStream << std::fixed << std::setprecision(0) << Xcount;
	CmdStr = PreStrng + raStream.str() + ";";
	SendCommand(CmdStr);

	// Back to tracking
	CmdStr = "AX JF" + FormatDouble(TrkRate, 10, 6) + ";";
	SendCommand(CmdStr);

	// DEC position load
	if (Ycount < 0.0)
		PreStrng = "AY LP-";
	else
		PreStrng = "AY LP";

	Ycount = std::abs(Ycount);

	std::ostringstream decStream;
	decStream << std::fixed << std::setprecision(0) << Ycount;
	CmdStr = PreStrng + decStream.str() + ";";
	SendCommand(CmdStr);

	coord->RASync = 0.0;
	coord->DecSync = 0.0;

	UpdateCoord();
}